src/repr/type_binary.ml

```diff
@@ -224,12 +225,12 @@ module Decode = struct
     (ofs, Int64.float_of_bits f)
 
   let int buf ofs =
-    let rec aux n p ofs =
+    let rec aux buf n p ofs =
```

Comment:
Does passing `buf` to `aux` make a difference here?

Reply:
I believe that it does not make a closure with an free-variable.

Reply:
yes I haven't checked the generated code but it should generate a loop if all the variable are bound as function parameters (might be worth checking)

Reply:
Thanks for the clarification!